pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/BaseSegmentAssignment.java

```diff
@@ -201,53 +170,10 @@ protected Map<String, Map<String, String>> reassignSegments(String instanceParti
     } else {
       int numReplicaGroups = instancePartitions.getNumReplicaGroups();
       int numPartitions = instancePartitions.getNumPartitions();
-
-      if (numReplicaGroups == 1 && numPartitions == 1) {
-        // Non-replica-group based assignment
-
-        List<String> instances =
-            SegmentAssignmentUtils.getInstancesForNonReplicaGroupBasedAssignment(instancePartitions, _replication);
-        newAssignment =
-            SegmentAssignmentUtils.rebalanceTableWithHelixAutoRebalanceStrategy(currentAssignment, instances,
-                _replication);
-      } else {
-        // Replica-group based assignment
-
-        checkReplication(instancePartitions);
-
-        if (_partitionColumn == null || numPartitions == 1) {
-          // NOTE: Shuffle the segments within the current assignment to avoid moving only new segments to the new added
-          //       servers, which might cause hotspot servers because queries tend to hit the new segments. Use the
-          //       table name hash as the random seed for the shuffle so that the result is deterministic.
-          List<String> segments = new ArrayList<>(currentAssignment.keySet());
-          Collections.shuffle(segments, new Random(_tableNameWithType.hashCode()));
-
-          newAssignment = new TreeMap<>();
-          SegmentAssignmentUtils.rebalanceReplicaGroupBasedPartition(currentAssignment, instancePartitions, 0, segments,
-              newAssignment);
-        } else {
-          Map<Integer, List<String>> instancePartitionIdToSegmentsMap =
-              getInstancePartitionIdToSegmentsMap(currentAssignment.keySet(), instancePartitions.getNumPartitions());
-
-          // NOTE: Shuffle the segments within the current assignment to avoid moving only new segments to the new added
-          //       servers, which might cause hotspot servers because queries tend to hit the new segments. Use the
-          //       table name hash as the random seed for the shuffle so that the result is deterministic.
-          Random random = new Random(_tableNameWithType.hashCode());
-          for (List<String> segments : instancePartitionIdToSegmentsMap.values()) {
-            Collections.shuffle(segments, random);
-          }
-
-          return SegmentAssignmentUtils.rebalanceReplicaGroupBasedTable(currentAssignment, instancePartitions,
-              instancePartitionIdToSegmentsMap);
-        }
-      }
+      // determine segment assignment strategy
+      setSegmentAssignmentStrategy(numReplicaGroups, numPartitions);
```

Comment:
This would happen in Loop as every tierInstancePartitions will have different assignment and assignment strategy